src/Core/Services/Implementations/UserService.cs

```diff
@@ -626,8 +626,59 @@ public override Task<IdentityResult> ChangePasswordAsync(User user, string maste
             return IdentityResult.Success;
         }
         
-        public async Task<IdentityResult> AdminResetPasswordAsync(User user, string newMasterPassword, string key)
+        public async Task<IdentityResult> AdminResetPasswordAsync(OrganizationUserType callingUserType, Guid orgId, Guid id, string newMasterPassword, string key)
         {
+            // Org must be able to use reset password
+            var org = await _organizationRepository.GetByIdAsync(orgId);
+            if (org == null || !org.UseResetPassword)
+            {
+                throw new BadRequestException("Organization does not allow password reset.");
+            }
+            
+            // Enterprise policy must be enabled 
+            var resetPasswordPolicy =
+                await _policyRepository.GetByOrganizationIdTypeAsync(orgId, PolicyType.ResetPassword);
+            if (resetPasswordPolicy == null || !resetPasswordPolicy.Enabled)
+            {
+                throw new BadRequestException("Organization does not have the password reset policy enabled.");
+            }
+            
+            // Org User must be confirmed and have a ResetPasswordKey
+            var orgUser = await _organizationUserRepository.GetByIdAsync(id);
```

Comment:
Since you are looking up the orgUser in here, I dont think you need to pass in the order user type too

Reply:
The `OrganizationUserType` being passed in as a parameter to the method is being taken from the `currentContext` and is different than the `orgUser` being looked up here. `orgUser` is the person who's password is being changed, while the `callingUserType` is the admin making the request.